lib/improver/spotdata/tests/test_neighbour_finding.py

```diff
@@ -0,0 +1,679 @@
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the spotdata.NeighbourFinding plugin."""
+
+
+import unittest
+
+from iris.coords import DimCoord
+from iris.cube import Cube
+from iris.tests import IrisTest
+from collections import OrderedDict
+import numpy as np
+
+from improver.spotdata.neighbour_finding import PointSelection
+
+
+class TestNeighbourFinding(IrisTest):
+
+    """Test the neighbour finding plugin."""
+
+    def setUp(self):
+        """Create a cube containing a regular lat-lon grid."""
+        data = np.zeros((20, 20))
+        latitudes = np.linspace(-90, 90, 20)
+        longitudes = np.linspace(-180, 180, 20)
+        latitude = DimCoord(latitudes, standard_name='latitude',
+                            units='degrees')
+        longitude = DimCoord(longitudes, standard_name='longitude',
+                             units='degrees')
+
+        cube = Cube(data,
+                    long_name="test_data",
+                    dim_coords_and_dims=[(latitude, 0), (longitude, 1)],
+                    units="1")
+
+        orography = cube.copy()
+        orography.rename('surface_altitude')
+        land = cube.copy()
+        land.rename('land_binary_mask')
+        land.data = land.data + 1
+
+        ancillary_data = {}
+        ancillary_data.update({'orography': orography})
+        ancillary_data.update({'land': land})
+
+        sites = OrderedDict()
+        sites.update({'100': {'latitude': 50,
+                              'longitude': 0,
+                              'altitude': 10,
+                              'gmtoffset': 0
+                              }
+                      })
+
+        neighbour_list = np.empty(1, dtype=[('i', 'i8'),
+                                            ('j', 'i8'),
+                                            ('dz', 'f8'),
+                                            ('edge', 'bool_')])
+
+        self.cube = cube
+        self.ancillary_data = ancillary_data
+        self.sites = sites
+        self.neighbour_list = neighbour_list
+
+    def return_types(self, method):
```

Comment:
No. Then when each method uses them it will be e.g. test_check_return_types, which is getting a bit silly. As is stands, each of the actually run tests reads e.g. test_return_types.... which I think makes sense if any error messages arise.